openstack/standalone-cinder/pkg/volumeservice/actions.go

```diff
@@ -108,11 +109,26 @@ func CreateCinderVolume(vs *gophercloud.ServiceClient, options controller.Volume
 // become available.  The connection information cannot be retrieved from cinder
 // until the volume is available.
 func WaitForAvailableCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error {
-	// TODO: Implement proper polling instead of brain-dead timers
 	c := make(chan error)
-	go time.AfterFunc(5*time.Second, func() {
-		c <- nil
-	})
+	go func() error {
+		num :=0
+		for {
+			time.Sleep(3 * time.Second)
+			if num > 10 {
+				err := errors.New("Failed to get volume status after max retries")
```

Comment:
I would prefer at least a pair of constants to define the sleep interval and max retries

Reply:
absolutely, I will fix it in next patch